### PR TITLE
Use normalization from training set on evaluation

### PR DESCRIPTION
**Pull request by daviddanan**
- Change normalization for evaluation, use training dataset for that, 
- Change threshold for std in scaler

### CODE REVIEW

File: lips/benchmark/airfransBenchmark.py [hunk 1]
```diff
@@ -164,6 +166,15 @@ def evaluate_simulator(self,
...
         """
         self.augmented_simulator = augmented_simulator
+
+        field_names = self.train_dataset._attr_y
+        chunk_sizes = [int(simulation[1]) for simulation in self.train_dataset.extra_data["simulation_names"]]
+        flattened_train = np.concatenate([self.train_dataset.data[field_name][:, None] for field_name in field_names], axis = 1)
+        mean_observ,std_observ = iterative_fit(flattened_train,chunk_sizes)
+        self.ml_normalization["mean"] = mean_observ
+        self.ml_normalization["std"] = std_observ
+        self.evaluation.ml_normalization = self.ml_normalization
```
Comment (Mleyliabadi) [inline lips/benchmark/airfransBenchmark.py:176]: Following my comments in `airfransEvaluation`, for the sake of generality, you could pass the ml_normalization as an argument to the evaluation instead of changing the class attribute. 

For sure, it is just my idea, I let you decide the way you prefer.

File: lips/evaluation/airfrans_evaluation.py [hunk 2]
```diff
@@ -123,14 +124,20 @@ def evaluate_ml(self) -> dict:
...
         #Normalize prediction and observation before computing mlMetrics
-        chunk_sizes = [int(simulation[1]) for simulation in self.observation_metadata["simulation_names"]]
 
         field_names = self.predictions.keys()
-        flattened_observation = np.concatenate([self.observations[field_name][:, None] for field_name in field_names], axis = 1)
-        mean_observ,std_observ = iterative_fit(flattened_observation,chunk_sizes)
+        if self.ml_normalization is None:
+            self.logger.info("Using current dataset to normalize")
+            chunk_sizes = [int(simulation[1]) for simulation in self.observation_metadata["simulation_names"]]
+            flattened_observation = np.concatenate([self.observations[field_name][:, None] for field_name in field_names], axis = 1)
```
Comment (Mleyliabadi) [inline lips/evaluation/airfrans_evaluation.py:132]: I think the `self.observations` here, refers to the ground truth test dataset passed through the evaluation call. However, as your aim is to fit the scaler parameters on train dataset and use them to normalize the test dataset, I think this does not make sense.

**Review by Mleyliabadi — CHANGES_REQUESTED** [review]
Some comments in the code. Let me know what do you think about and we merge the PR just after.
